tutorials/working-in-charts.qmd

```diff
@@ -0,0 +1,179 @@
+
+Next we solve the vector transport problem `solve_for([θₚ, φₚ], [θₓ, φₓ], [θy, φy])`, sample the result at the selected time steps and store the result in `geo`. The solution includes the geodesic which we extract and convert to a sequence of points digestible by `Makie.jl`, `geo_ps`.
+`[θₚ, φₚ]` is the parametrization in chart (0, 0) of the starting point of the geodesic.
+The direction of the geodesic is determined by `[θₓ, φₓ]`, coordinates of the tangent vector at the starting point expressed in the induced basis of chart `i_p0x` (which depends on the initial point).
+Finally, `[θy, φy]` are the coordinates of the tangent vector that will be transported along the geodesic, which are also expressed in same basis as `[θₓ, φₓ]`.
+
+We won't draw the transported vector at every point as there would be too many arrows, which is why we select every 100th point only for that purpose with `pt_indices`. Then, `geo_ps_pt` contains points at which the transported vector is tangent to and `geo_Ys` the transported vector at that point, represented in the embedding.
+
+The logarithmic map will be solved between points with parametrization `bvp_a1` and `bvp_a2` in chart `bvp_i`.
+The result is assigned to variable `bvp_sol` and then sampled with time step 0.05. The result of this sampling is converted from parameters in chart `bvp_i` to point in the embedding and stored in `geo_r`.
+
+
+```{julia}
+function solve_for(p0x, X_p0x, Y_transp, T)
+    p = [Manifolds._torus_param(M, p0x...)...]
+    i_p0x = Manifolds.get_chart_index(M, A, p)
+    p_exp = Manifolds.solve_chart_parallel_transport_ode(
+        M,
+        [0.0, 0.0],
+        X_p0x,
+        A,
+        i_p0x,
+        Y_transp;
+        final_time=T,
+    )
+    return p_exp
+end;
+```
+
+
+### Solving parallel Transport ODE
+
+We set the end time `t_end` and time step `dt`.
+
+```{julia}
+t_end = 2.0
+dt = 1e-1
+```
+
+We also parametrise the start point and direction.
+
+```{julia}
+θₚ = π/10
+φₚ = -π/4
+θₓ = π/2
+φₓ = 0.7
+θy = 0.2
+φy = -0.1
+
+geo = solve_for([θₚ, φₚ], [θₓ, φₓ], [θy, φy], t_end)(0.0:dt:t_end);
+# geo_ps = [Point3f(s[1]) for s in geo]
+# pt_indices = 1:div(length(geo), 10):length(geo)
+# geo_ps_pt = [Point3f(s[1]) for s in geo[pt_indices]]
+# geo_Ys = [Point3f(s[3]) for s in geo[pt_indices]]
+
+# ax1, fig1 = torus_figure()
+# arrows!(ax1, geo_ps_pt, geo_Ys, linewidth=0.05, color=:blue)
+# lines!(geo_ps; linewidth=4.0, color=:green)
+# fig1
+```
+
+![fig-pt](working-in-charts/working-in-charts-transport.png)
+
+### Solving the logairthmic map ODE
+
+```{julia}
+θ₁=π/2
+φ₁=-1.0
+θ₂=-π/8
+φ₂=π/2
+
+bvp_i = (0, 0)
+bvp_a1 = [θ₁, φ₁]
+bvp_a2 = [θ₂, φ₂]
+bvp_sol = Manifolds.solve_chart_log_bvp(M, bvp_a1, bvp_a2, A, bvp_i);
+# geo_r = [Point3f(get_point(M, A, bvp_i, p[1:2])) for p in bvp_sol(0.0:0.05:1.0)]
+
+# ax2, fig2 = torus_figure()
+# lines!(geo_r; linewidth=4.0, color=:green)
+# fig2
+```
+
+![fig-geodesic](working-in-charts/working-in-charts-geodesic.png)
```

Comment:
You could add a final paragraph that mentions the (old) `.jl` file which is an interactive (Pluto notebook) version of this tutorial?